shuup/front/apps/simple_search/views.py

```diff
@@ -67,9 +67,11 @@ def get_search_product_ids(request, query, limit=150):
     if product_ids is None:
         entry_query = get_compiled_query(
             query, ['sku', 'translations__name', 'translations__description', 'translations__keywords'])
-        product_ids = list(Product.objects.filter(entry_query).distinct().values_list("pk", flat=True))[:limit]
+        product_ids = list(
+            Product.objects.searchable(shop=request.shop).filter(entry_query).distinct().values_list("pk", flat=True)
+        )
         cache.set(cache_key, product_ids, 60 * 5)
-    return product_ids
+    return product_ids[:limit]
```

Comment:
You cache all but return the limited ones hence changing the original implementation, is this intented?